src/shared/Core/ApplicationBase.cs

```diff
@@ -75,6 +77,31 @@ public Task<int> RunAsync(string[] args)
                 Context.Trace.WriteLine("Tracing of secrets is enabled. Trace output may contain sensitive information.");
             }
 
+            // Enable TRACE2 tracing
+            if (Context.Settings.GetTrace2TracingEnabled(out string trace2Value))
+            {
+                var connectionPath = PlatformUtils.IsWindows() ? trace2Value.TrimUntilLastIndexOf("/") :
+                    trace2Value.TrimUntilIndexOf(":");
```

Comment:
On Windows, we should strip off the UNC host and drive -- so basically strip off the "//./pipe/" prefix.  In this context the "." is the UNC hostname and "pipe" is the mount point -- since all names pipes are created on the special named pipe file system (NPFS) pseudo-drive.  Named pipes can have more slashes (but usually don't).
It's the prefix that is the key.

On Unix, we want to strip off the "af_unix:", "af_unix:stream:" or "af_unix:dgram:" prefix.  (The optional stream or dgram term specifies TCP or UDP and if omitted we should try both types of connections.). (I'm not sure the GOLANG daemon supports both protocols right now, but we should be able to parse the it here.)

Reply:
Thanks for bringing these additional use cases to my attention. It's a perfect candidate for a unit test, let me put some more thought into how we can implement that.